### PR TITLE
Themes: Remove blank menu item in popover menu for theme cards

### LINKED ISSUE #42052: [Themes page]Theme menu has a blank menu item
#### Steps to reproduce

1. Starting at URL: https://wordpress.com/themes/
2. Click on the three dotted button and check underneath the menu items
3. You'll be able to hover over an empty one which, on some themes, acts like an activation button.

#### Browser / OS version
Chrome 81

#### Screenshot / Video
https://monosnap.com/file/QvNH8eJjH5Ru9Mq4ezs5XyLsqRDQ70

#### Context / Source
#manual-testing
Confirmed by @kjellr here: p1589197758250600-slack

### PR DESCRIPTION
**Pull request by DanielSmithMichigan**
| Before | After |
| --- | --- |
| image | image |

Removes this [blank menu item](https://d.pr/v/adaYsx).

The blank menu item dispatches the `showAutoLoadingHomepageWarning` action, but this is the same action that the ["activate" button dispatches](https://github.com/Automattic/wp-calypso/blob/3c990841748ed80ca1de8663b6cf0cebf2346a27/client/state/themes/actions/activate.js#L48). Since they're both part of the same menu, this leads me to believe that the blank menu item is a duplicate and can be removed.

While the blank menu item does create the "showAutoLoadingHomepage" modal, it looks like the "activate" button does this as well, so it seems like we don't need the blank menu item. Note that both buttons have [very similar functionality](https://user-images.githubusercontent.com/10274366/118847112-66daf780-b89b-11eb-929b-21d28c139ec3.png).

Related to #42052

retrofox has already reviewed/tested, but I'm just noting my thoughts here, now that I've done my due diligence, to show what I've tested as well.

### Testing Notes

The "theme-options" component is used in many places. I personally did a thorough review to determine that the only place which uses this showAutoLoadingHomepageWarning option from that theme-options component is the menu shown above. Admittedly, searching for usage of this option is a part of the QA process for this pull request, so you will either need to take my investigation for granted or repeat this large effort (which took me one whole day).

Assuming we take that investigation for granted, then all we need to do is look at the popover menu in each place that it is used.

### Testing Instructions

1. Check the theme showcase for:
- A simple site
- An atomic site
- A non-atomic site, connected via jetpack

Be sure to check the recommended and advanced theme sections separately.

Expected Behavior: All the previously existing menu options (aside from the blank one) should still exist, and should function like they did before.

2. When activating a theme on a simple site, check that the autoloading warning modal shows, both from the theme showcase and from the single theme page.

### CODE REVIEW

**Comment by retrofox** [comment]
~First question. I'm not seeing the warning modal about changing the homepage anymore. Do you know whether it's intentional?~

Sorry, testing with the wrong site type.

**Review by retrofox — APPROVED** [review]
> The blank menu item dispatches the showAutoLoadingHomepageWarning action, but this is the same action that the "activate" button dispatches. 

:this: is correct.

I've tested and it works as expected:

Step | before | after
------|-------|-----
The additional empty row is gone | image | image
It triggers the warning modal when activating the theme | image | image

Same behavior when activating the theme from the single page:

`http://calypso.localhost:3000/theme/twentytwentyone/retrofocs.wordpress.com`

before to click on activate | after
-----|-----
image | image

Thanks, @DanielSmithMichigan for this fix. LGTM.

**Comment by DanielSmithMichigan** [comment]
I traced every path in the code that interacts with the showAutoLoadingHomepageWarning option

tl;dr [it only shows up in theme popover menus](https://user-images.githubusercontent.com/10274366/119528792-823c7b80-bd4f-11eb-82e3-a12d39315bba.png)

**Comment by DanielSmithMichigan** [comment]
Where options are featured:

[After uploading a theme](https://user-images.githubusercontent.com/10274366/119534330-c7af7780-bd54-11eb-81f6-c690cb6c3b89.png)

[Popover Menus](https://user-images.githubusercontent.com/10274366/119534430-e6157300-bd54-11eb-822e-e5345a7a1a2e.png)

[Screenshot Options](https://user-images.githubusercontent.com/10274366/119534540-09d8b900-bd55-11eb-9cef-537245087c5e.png)

[Themes Page](https://user-images.githubusercontent.com/10274366/119534626-1f4de300-bd55-11eb-9ffc-ce8858852ba0.png)

[Theme Preview](https://user-images.githubusercontent.com/10274366/119582043-d49e8c00-bd91-11eb-926f-8d86d697b581.png)

**Comment by DanielSmithMichigan** [comment]
I executed on this test plan. I think it's good to merge, will do that tomorrow, and will upload some screenshots from testing tomorrow

image

**Comment by DanielSmithMichigan** [comment]
| Before | After |
| --- | --- |
| image | image |

https://user-images.githubusercontent.com/10274366/119681346-fb030c80-be0f-11eb-9536-69eea803a72f.mov

https://user-images.githubusercontent.com/10274366/119681239-e32b8880-be0f-11eb-9547-1047aab8dffb.mov

https://user-images.githubusercontent.com/10274366/119681259-e757a600-be0f-11eb-9586-bf5dcc49a1a5.mov
